tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -207,6 +208,17 @@ FORCE_INLINE
 void relay_to_next_cb(uint32_t data_ptr,
                       uint32_t length) {
 
+    static_assert(
+        preamble_size == 0 || preamble_size == sizeof(dispatch_packet_header_t),
+        "Dispatcher preamble size must be 0 or sizeof(dispatch_packet_header_t)");
+
+    // The downstream packetizing stage will initialize the other fields, but it needs info on
+    // the length of the transfer to be packetized.
+    if (preamble_size > 0) {
+        uint64_t downstream_noc_addr = get_noc_addr_helper(downstream_noc_xy, downstream_cb_data_ptr);
+        noc_inline_dw_write(downstream_noc_addr, length + preamble_size);
```

Comment:
on the prefetch side, I think I store the value to L1 then send it which is a data race, I'll look into that. prefetcher tends to be input bound (from pcie or dram) probably why we haven't hit it